tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,436 @@
+    f2 = np.zeros(bound_faces.size)
+    f3 = np.zeros(bound_faces.size)
+    f4 = np.zeros(bound_faces.size)
+    f5 = np.zeros(bound_faces.size)
+
+    u = np.zeros((g.dim, g.num_faces))
+
+    # Constant term in (7.4.5)
+    m = 1 / (4 * np.pi * (1 - poi))
+
+    # Second term in (7.4.5)
+    f2[:] = m * (
+        np.log(np.sqrt((xf[0, :] - h) ** 2 + xf[1] ** 2))
+        - np.log(np.sqrt((xf[0, :] + h) ** 2 + xf[1] ** 2))
+    )
+
+    # First term in (7.4.5)
+    f3[:] = -m * (
+        np.arctan2(xf[1, :], (xf[0, :] - h)) - np.arctan2(xf[1, :], (xf[0, :] + h))
+    )
+
+    # The following equalities can be found on page 91 where f3,f4 as equation (5.5.3) and ux, uy in (5.5.1)
+    # Also this is in the coordinate system described in (4.5.1) at page 57, which essentially is a rotation.
+    f4[:] = m * (
+        xf[1, :] / ((xf[0, :] - h) ** 2 + xf[1, :] ** 2)
+        - xf[1, :] / ((xf[0, :] + h) ** 2 + xf[1, :] ** 2)
+    )
+
+    f5[:] = m * (
+        (xf[0, :] - h) / ((xf[0, :] - h) ** 2 + xf[1, :] ** 2)
+        - (xf[0, :] + h) / ((xf[0, :] + h) ** 2 + xf[1, :] ** 2)
+    )
+
+    u[0, bound_faces] = du * (
+        -(1 - 2 * poi) * np.cos(alpha) * f2[:]
+        - 2 * (1 - poi) * np.sin(alpha) * f3[:]
+        - xf[1, :] * (np.cos(alpha) * f4[:] + np.sin(alpha) * f5[:])
+    )
+    u[1, bound_faces] = du * (
+        -(1 - 2 * poi) * np.sin(alpha) * f2[:]
+        + 2 * (1 - poi) * np.cos(alpha) * f3[:]
+        - xf[1, :] * (np.sin(alpha) * f4[:] - np.cos(alpha) * f5[:])
+    )
+
+    return u
+
+
+def assign_bem(
+    g: pp.Grid,
+    h: float,
+    bound_faces: np.ndarray,
+    theta: float,
+    bem_centers: np.ndarray,
+    u_a: np.ndarray,
+    poi: float,
+) -> np.ndarray:
+    """
+    Compute analytical displacement using the BEM method for the Sneddon problem.
+
+    Parameter
+        g: Subdomain grid
+        h: bem segment length
+        bound_faces: boundary faces
+        theta: fracture orientation
+        bem_centers: bem segments centers
+        u_a: Sneddon's analytical relative normal displacement
+        poi: Poisson ratio
+
+    Return:
+        Semi-analytical boundary displacement values
+    """
+
+    bc_val = np.zeros((g.dim, g.num_faces))
+
+    alpha = np.pi / 2 - theta
+
+    bound_face_centers = g.face_centers[:, bound_faces]
+
+    for i in range(0, u_a.size):
+
+        new_bound_face_centers = transform(bem_centers[:, i], bound_face_centers, alpha)
+
+        u_bound = get_bc_val(
+            g, bound_faces, new_bound_face_centers, h, poi, alpha, u_a[i]
+        )
+
+        bc_val += u_bound
+
+    return bc_val
+
+
+
+
+class ManuExactSneddon2dSetup:
+    """
+    Class for setting up the analytical solution for the pressurized fracture problem.
+    
+    
+    """
+  
+    def __init__(self, setup):
+        # Initialize private variables from the setup dictionary
+        self.p0 = setup.get("p0")
```

Comment:
Why not just use setup.p0 etc.? We could also access as self.setup.p0 in the rest of this class.